File: travel_api_gateway/src/main/java/com/example/travel_api_gateway/context/CustomSecurityContextHolder.java
package com.example.travel_api_gateway.context;

public class CustomSecurityContextHolder {
    private static final ThreadLocal<CustomSecurityContext> contextHolder = new ThreadLocal<>();

    public static void setContext(CustomSecurityContext context) {
        contextHolder.set(context);
    }

    public static CustomSecurityContext getContext() {
        return contextHolder.get();
    }

    public static void clearContext() {
        contextHolder.remove();
    }
}

File: travel_reservation/src/main/kotlin/com/example/travel_reservation/context/CustomSecurityContextHolder.kt
package com.example.travel_reservation.context

object CustomSecurityContextHolder {
    private val contextHolder = ThreadLocal<CustomSecurityContext>()

    fun setContext(context: CustomSecurityContext) {
        contextHolder.set(context)
    }

    fun getContext(): CustomSecurityContext? {
        return contextHolder.get()
    }

    fun clearContext() {
        contextHolder.remove()
    }
}
